lark/visitors.py

```diff
@@ -136,7 +138,7 @@ def transform(self, tree: Tree[_Leaf_T]) -> _Return_T:
         "Transform the given tree, and return the final result"
         return self._transform_tree(tree)
 
-    def __mul__(self, other: 'Transformer[_Return_T, _Leaf_T]') -> 'TransformerChain[_Return_T, _Leaf_T]':
+    def __mul__(self: 'Transformer[Tree[_Leaf_U], _Leaf_T]', other: 'Transformer[_Return_V, _Leaf_U]') -> 'TransformerChain[_Return_V, _Leaf_T]':
```

Comment:
Reminder from an earlier conversation:
> I was about to add a new comment about that. I agree that it reads better at `Transformer[_Leaf_T, _Return_T] `. I initially added the leaf type at the end to minimize that changes. However, going from 1 to 2 generic parameters is a breaking change anyways, so moving leaf to be the first type passed in isn't a huge issue.
> 
> Note: These typing changes are only breaking changes for for code already taking advantage of type annotations and the generic types.

Since the PR didn't make it into the 1.0 release, that is something that should be kept in mind. If the choice is made to go the 2.0 route, I remember seeing a couple of places where minor API changes could resolve a few of the mypy errors that are currently reported on the library.